### PR TITLE
TM-315- Show and hide the bulk rejection Continue button on applicat…

### PR DESCRIPTION
**Pull request by jawwadbaig**
#TM-315- Show and hide the bulk rejection Continue button on applications pages based on applications status.

### CODE REVIEW

File: src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Orchestrators/PledgeOrchestratorTests.cs [hunk 1]
```diff
@@ -162,6 +162,52 @@ public async Task GetApplications_Returns_Valid_ViewModel()
             });
         }
 
+        [Test]
+        public async Task GetApplications_With_Status_Pending_Renders_Continue_Button()
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Orchestrators/PledgeOrchestratorTests.cs:166]: I'd be tempted to fold these two tests into a single test with multiple test cases - you could pass in parameters for:
a) how many pending applications there are
b) the expected "renderbutton" boolean

You could then use test cases for 0,1 and 2 pending applications.

And then maybe add an additional test to prove that applications of other statuses do not result in the button being rendered

File: src/SFA.DAS.LevyTransferMatching.Web/Models/Pledges/ApplicationsViewModel.cs [hunk 2]
```diff
@@ -12,5 +12,6 @@ public class ApplicationsViewModel : ApplicationsPostRequest
         public bool DisplayRejectedBanner { get; set; }
         public string RejectedEmployerName { get; set; }
         public bool RenderCreatePledgeButton { get; set; }
+        public bool RenderContinueButton { get; set; }
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Web/Models/Pledges/ApplicationsViewModel.cs:15]: I would call this "RenderRejectButton" or similar, for clarity

File: src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs [hunk 3]
```diff
@@ -215,7 +215,8 @@ public async Task<ApplicationsViewModel> GetApplications(ApplicationsRequest req
                 UserCanClosePledge = result.PledgeStatus != PledgeStatus.Closed && isOwnerOrTransactor,
                 EncodedPledgeId = request.EncodedPledgeId,
                 RenderCreatePledgeButton = isOwnerOrTransactor,
-                Applications = viewModels
+                Applications = viewModels,
+                RenderContinueButton = viewModels.Where(x => x.Status == ApplicationStatus.Pending).ToList().Count() >= 1
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs:219]: I think you could use the `Any` method here and be far more concise.

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Web/Models/Pledges/ApplicationsViewModel.cs (hunk 2 above)
Comment (jawwadbaig) [reply src/SFA.DAS.LevyTransferMatching.Web/Models/Pledges/ApplicationsViewModel.cs:15]: This is now changed to "RenderRejectButton"

File: src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs (hunk 3 above)
Comment (jawwadbaig) [reply src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/PledgeOrchestrator.cs:219]: yes sure, this is now updated and condition looks neat

File: src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Orchestrators/PledgeOrchestratorTests.cs (hunk 1 above)
Comment (jawwadbaig) [reply src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Orchestrators/PledgeOrchestratorTests.cs:166]: Created two tests and added multiple test cases to test the number of pending applications and behaviors when non-pending applications are passed.

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**